examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt

```diff
@@ -64,15 +65,15 @@ class ReplayHistoryActivity : AppCompatActivity() {
     // You choose your loading mechanism. Use Coroutines, ViewModels, RxJava, Threads, etc..
     private var loadNavigationJob: Job? = null
 
+    @SuppressLint("MissingPermission")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_replay_history_layout)
         mapView.onCreate(savedInstanceState)
 
         selectHistoryButton.setOnClickListener {
             val activityIntent = Intent(this, HistoryFilesActivity::class.java)
-            startActivity(activityIntent)
-            finish()
+            startActivityForResult(activityIntent, HistoryFilesActivity.REQUEST_CODE)
```

Comment:
Updating the example to use `startActivityForResult` to showcase the `resetTripSession` function.

When we start an activity for result, the current activity is never destroyed. Making it so, when the onActivityResult comes back it will be able to update the state of MapboxNavigation without completely restoring it.